syntax = "proto3";

package xxdb;

message AuthRequest {
    string dbname = 1;
    string payload = 2;
}

message CommonRequest {
    enum Command {
        HEARTBEAT = 0;
        GET = 3;
        PUT = 4;
        BULK_PUT = 5;
    }

    message GetRequest {
        string key = 1;
    }

    message PutRequest {
        string key = 1;
        bytes value = 2;
    }

    Command command = 1;  // required
    AuthRequest auth_payload = 2;
    GetRequest get_payload = 3;
    PutRequest put_payload = 4;
    repeated PutRequest bulkput_payload = 5;
}

message CommonResponse {
    enum Status {
        OK = 0;
        FAILED = 1;
        ERROR = 2;
    }
    Status status = 1;
    string auth_payload = 2;
    string error_payload = 3;
    bytes get_payload = 4;
}
